src/main/java/org/broadinstitute/hellbender/tools/exome/sexgenotyper/PloidyAnnotatedTargetCollection.java

```diff
@@ -0,0 +1,219 @@
+package org.broadinstitute.hellbender.tools.exome.sexgenotyper;
+
+import com.google.common.collect.Sets;
+import htsjdk.samtools.util.Locatable;
+import org.broadinstitute.hellbender.exceptions.UserException;
+import org.broadinstitute.hellbender.tools.exome.Target;
+import org.broadinstitute.hellbender.tools.exome.TargetCollection;
+import org.broadinstitute.hellbender.utils.IndexRange;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+
+import javax.annotation.Nonnull;
+import java.util.*;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+/**
+ * A collection of {@link Target} instances along with helper methods for generating their genotype ploidy
+ * annotations on-the-fly using provided contig ploidy annotations.
+ *
+ * @author Mehrtash Babadi &lt;mehrtash@broadinstitute.org&gt;
+ */
+public final class PloidyAnnotatedTargetCollection implements TargetCollection<Target> {
+
+    /**
+     * Map from targets to their ploidy annotations (based on target contigs)
+     */
+    private Map<Target, ContigPloidyAnnotation> targetToContigPloidyAnnotationMap;
+
+    /**
+     * List of autosomal targets
+     */
+    private List<Target> autosomalTargetList;
+
+    /**
+     * List of allosomal targets
+     */
+    private List<Target> allosomalTargetList;
+
+    /**
+     * List of all targets
+     */
+    private List<Target> fullTargetList;
+
+    /**
+     * Set of all targets
+     */
+    private Set<Target> fullTargetSet;
+
+
+    /**
+     * Public constructor.
+     *
+     * @param contigAnnotsList list of contig ploidy annotations.
+     * @param targetList list of targets
+     */
+    public PloidyAnnotatedTargetCollection(@Nonnull final List<ContigPloidyAnnotation> contigAnnotsList,
+                                           @Nonnull final List<Target> targetList) {
+        performValidityChecks(targetList, contigAnnotsList);
+
+        fullTargetList = Collections.unmodifiableList(targetList);
+        fullTargetSet = Collections.unmodifiableSet(new HashSet<>(fullTargetList));
+
+        /* map targets to ploidy annotations */
+        final Map<String, ContigPloidyAnnotation> contigNameToContigPloidyAnnotationMap = contigAnnotsList.stream()
+                .collect(Collectors.toMap(ContigPloidyAnnotation::getContigName, Function.identity()));
+        targetToContigPloidyAnnotationMap = Collections.unmodifiableMap(
+                fullTargetList.stream().collect(Collectors.toMap(Function.identity(),
+                        target -> contigNameToContigPloidyAnnotationMap.get(target.getContig()))));
```

Comment:
They shouldn't be: parentheses and require additional tabs after line break. I am basically listening to IntelliJ for line breaks.